frontend/src/scenes/events-management/live-events/liveEventsTableLogic.ts

```diff
@@ -0,0 +1,182 @@
+            {
+                setFilters: (state, { filters }) => ({ ...state, ...filters }),
+            },
+        ],
+        clientSideFilters: [
+            {},
+            {
+                setClientSideFilters: (_, { clientSideFilters }) => clientSideFilters,
+            },
+        ],
+        eventsSource: [
+            null,
+            {
+                updateEventsSource: (_, { source }) => source,
+            },
+        ],
+        streamPaused: [
+            false,
+            {
+                pauseStream: () => true,
+                resumeStream: () => false,
+            },
+        ],
+        curEventProperties: [
+            [],
+            {
+                setCurEventProperties: (_, { curEventProperties }) => curEventProperties,
+            },
+        ],
+        stats: [
+            { users_on_product: null },
+            {
+                setStats: (_, { stats }) => stats,
+            },
+        ],
+    }),
+    selectors(({ selectors }) => ({
+        eventCount: [() => [selectors.events], (events: any) => events.length],
+        filteredEvents: [
+            (s) => [s.events, s.clientSideFilters],
+            (events, clientSideFilters) => {
+                return events.filter((event) => {
+                    return Object.entries(clientSideFilters).every(([key, value]) => {
+                        return event[key] === value
+                    })
+                })
+            },
+        ],
+    })),
+    listeners(({ actions, values }) => ({
+        setFilters: () => {
+            actions.clearEvents()
+            actions.updateEventsConnection()
+        },
+        updateEventsConnection: async () => {
+            if (values.eventsSource) {
+                // @ts-expect-error
+                values.eventsSource.close()
+            }
+
+            if (values.streamPaused) {
+                return
+            }
+
+            if (!values.currentTeam) {
+                return
+            }
+
+            const { eventType } = values.filters
+            const url = new URL(`${liveEventsHostOrigin()}/events`)
+            url.searchParams.append('teamId', '2')
+            if (eventType) {
+                url.searchParams.append('eventType', eventType)
+            }
+
+            // @ts-expect-error
+            const source = new window.EventSourcePolyfill(url.toString(), {
+                headers: {
+                    Authorization: `Bearer ${values.currentTeam?.live_events_token}`,
+                },
+            })
+
+            const batch: Record<string, any>[] = []
+            source.onmessage = function (event: any) {
+                const eventData = JSON.parse(event.data) // Assuming the event data is in JSON format
+                batch.push(eventData)
+                if (batch.length >= 5) {
+                    // Process in batches of 5
+                    actions.addEvents(batch)
+                    batch.length = 0
+                }
```

Comment:
Batch size doesn't seem like the most important criterion – time since last batch is probably more generally useful (e.g. will work if you're developing and have very low volume). So I propose something like this:
**Call `addEvents` if last call was more than 200 ms ago, or if the batch is larger than 5**
Unsure about the exact numbers, but we can tweak that later